### PR TITLE
refactor: NFTokenAcceptOffer

### PR DESCRIPTION
**Pull request by connorjchen**
### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### TypeScript/Hooks Update

- [x] Updated files to React Hooks
- [x] Updated files to TypeScript

### CODE REVIEW

File: src/containers/shared/components/Transaction/NFTokenAcceptOffer/test/NFTokenAcceptOfferSimple.test.jsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      </I18nextProvider>,
+    )
+    expect(wrapper.find('[data-test="token-id"] .value')).toHaveText(
+      '000800006203F49C21D5D6E022CB16DE3538F248662FC73C29ABA6A90000000D',
+    )
+    expect(wrapper.find('[data-test="offer-id"] .value')).toHaveText(
+      '8278760A246D4464EE701D503091B9DB0D9790DD2BBE9CAABCA45B04A1A25B6B',
+    )
+    expect(wrapper.find('[data-test="amount"] .value')).toHaveText(
+      '0.0001 XRP',
```
Comment (ckniffen) [inline src/containers/shared/components/Transaction/NFTokenAcceptOffer/test/NFTokenAcceptOfferSimple.test.jsx:30]: You should add the utf-8 character code instead.  Many of the other tests do that already.

File: src/containers/shared/components/Transaction/NFTokenAcceptOffer/parser.ts [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
+export const parser: TransactionParser<
+  NFTokenAcceptOffer,
+  NFTokenAcceptOfferInstructions
+> = (tx, meta) => {
   const acceptedOfferNode = meta.AffectedNodes.find(
-    (node) => node.DeletedNode?.LedgerEntryType === 'NFTokenOffer',
+    (node: any) => node.DeletedNode?.LedgerEntryType === 'NFTokenOffer',
   )?.DeletedNode?.FinalFields
 
-  const acceptedOfferIDs = [tx.NFTokenBuyOffer, tx.NFTokenSellOffer].filter(
```
Comment (ckniffen) [inline src/containers/shared/components/Transaction/NFTokenAcceptOffer/parser.ts:8]: Does type checking not like this.

File: src/containers/shared/components/Transaction/NFTokenAcceptOffer/Simple.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        <SimpleRow label={t('offer_index')} className="dt" data-test="offer-id">
+          {offer}
+        </SimpleRow>
+      ))}
+      {amount && seller && buyer && tokenID && (
+        <>
+          <SimpleRow label={t('seller')} className="account" data-test="seller">
+            <Account account={seller} />
+          </SimpleRow>
+          <SimpleRow label={t('buyer')} className="account" data-test="buyer">
```
Comment (ckniffen) [inline src/containers/shared/components/Transaction/NFTokenAcceptOffer/Simple.tsx:29]: i dont think the className needs to be set on these two.  The className is already set inside `<Account>` and putting it on the row doesn't appear to do anything looking at the styles.

File: src/containers/shared/components/Transaction/NFTokenAcceptOffer/Simple.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            <Account account={buyer} />
+          </SimpleRow>
+          <SimpleRow label={t('token_id')} className="dt" data-test="token-id">
```
Comment (ckniffen) [inline src/containers/shared/components/Transaction/NFTokenAcceptOffer/Simple.tsx:32]: I think I told you incorrectly to use dt className on SimpleRow.  The css says it should be inside `.row` to have styles applied.  The name dt is not great (I realize it has been around for a while).  Originally it was used for showing `DestinationTag`s.  Maybe the class name should be changed to `long`.  And not set the different gray color. @mvadari Thoughts?

**Review by ckniffen — CHANGES_REQUESTED** [review]

File: src/containers/shared/components/Transaction/NFTokenAcceptOffer/parser.ts (hunk 2 above)
Comment (connorjchen) [reply src/containers/shared/components/Transaction/NFTokenAcceptOffer/parser.ts:8]: Yes, cause [string|undefined, string|undefined].filter is still potentially undefined

File: src/containers/shared/components/Transaction/NFTokenAcceptOffer/Simple.tsx (hunk 4 above)
Comment (connorjchen) [reply src/containers/shared/components/Transaction/NFTokenAcceptOffer/Simple.tsx:32]: For this PR, what should be changed?

Comment (ckniffen) [reply src/containers/shared/components/Transaction/NFTokenAcceptOffer/Simple.tsx:31]: I will create a new ticket for the `dt` thing

Comment (ckniffen) [reply src/containers/shared/components/Transaction/NFTokenAcceptOffer/Simple.tsx:31]: You should still remove the "account" className

**Review by ckniffen — APPROVED** [review]

File: src/containers/shared/components/Transaction/NFTokenAcceptOffer/Simple.tsx (hunk 4 above)
Comment (ckniffen) [reply src/containers/shared/components/Transaction/NFTokenAcceptOffer/Simple.tsx:31]: Created #361

File: src/containers/shared/components/Transaction/NFTokenAcceptOffer/types.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { TransactionCommonFields } from '../types'
+
+export interface NFTokenAcceptOffer extends TransactionCommonFields {
+  NFTokenSellOffer?: string
+  NFTokenBuyOffer?: string
+  NFTokenBrokerFee?:
+    | string
+    | { currency: string; value: string; issuer: string }
```
Comment (JST5000) [inline src/containers/shared/components/Transaction/NFTokenAcceptOffer/types.ts:8]: Is there an `Amount` or `IssuedCurrencyAmount` type somewhere in the repo? (That's what this type is)

File: src/containers/shared/components/Transaction/NFTokenAcceptOffer/types.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+export interface NFTokenAcceptOfferInstructions {
+  acceptedOfferIDs: string[]
+  amount?: { currency: string; amount: number; issuer?: string }
```
Comment (JST5000) [inline src/containers/shared/components/Transaction/NFTokenAcceptOffer/types.ts:13]: Ditto

**Review by JST5000 — APPROVED** [review]
LGTM other than maybe one typing simplification if it exists already - if it doesn't you should add the `Amount` type to the `types` file. (You can see the exact type in xrpl.js by the name `IssuedCurrencyAmount` - I'm pretty sure it matches)

File: src/containers/shared/components/Transaction/NFTokenAcceptOffer/types.ts (hunk 5 above)
Comment (connorjchen) [reply src/containers/shared/components/Transaction/NFTokenAcceptOffer/types.ts:8]: These types exist in Transactions/XChainAddAttestation. @ckniffen These should probably moved to general types file in Transactions in future.

Comment (ckniffen) [reply src/containers/shared/components/Transaction/NFTokenAcceptOffer/types.ts]: so it get turned into a number by formatAmount.  Maybe there should be a type of FormattedAmount

File: public/locales/en-US/translations.json [hunk 7]
```diff
@@ -380,7 +380,6 @@
   "buy_offers":"Buy Offers",
   "sell_offers":"Sell Offers",
   "offer_index": "Offer ID",
-  "owner": "Owner's Account",
```
Comment (ckniffen) [inline public/locales/en-US/translations.json:383]: @shawnxie999 said that this is used in his NFT transactions PR.

**Pull request merged**